lib/sla.js

```diff
@@ -231,101 +305,251 @@ function originateChannel(client, inbound, bridge, trunkData) {
+   * @param {Object} event - the event object related to this hang up.
+   */
+  function outboundHungupEarly(event) {
+    updateState(client, bridge.name, states.IDLE);
+    defer.reject(new customError.CustomError('EarlyOutboundHungup',
+          'Outbound channel hungup before dialing'));
+  }
+
+  /**
+   * Function that gets called when a DTMF digit is received from the outbound
+   *   channel. Originates a channel to the specified extension at the trunk
+   *   in the configuration file when the '#' key is pressed
+   * @param {Object} event - the event object related to this DTMF input.
+   *   Contains the DTMF digit as well.
+   * @param {Object} channel - the channel that sent this DTMF
+   */
+  function dtmfReceived(event, channel) {
+    var digit = event.digit;
+    switch (digit) {
+      case '#':
+        channelToDial = client.Channel();
+
+        var originate = Q.denodeify(channelToDial.originate.bind(
+              channelToDial));
+        originate({endpoint: 'SIP/' + toDial + '@'  +
+                   trunkData[0], app: 'sla',
+                   appArgs: 'dialed',
+                   timeout: 10})
+          .catch(function(err) {
+            defer.reject(err);
+          });
+
+        updateState(client, bridge.name, states.RINGING);
+
+        channelToDial.once('StasisStart', dialedEnteredStasis);
+        channelToDial.once('ChannelDestroyed', dialedChannelHungup);
+        outbound.removeListener('ChannelHangupRequest', outboundHungupEarly);
+        outbound.once('ChannelHangupRequest', outboundHungup);
+        break;
+
+      default:
+        toDial += digit;
+        break;
+    }
+  }
 }
 
-/** 
- * Utility function for checking if the bridgeName exists in the configuration
- *   file (is valid).
- * @param {String} configurationFile - the path and filename associated with the
- *   cofiguration file 
- * @param {String} extension - the name of the extension (and trunk so far)
- *   to be created/used
+/**
+ * Utility function for checking if a channel is a member of an extension
+ * @param {String} extension - the extension in question
+ * @param {String} channel - the channel in question
  * @return {Q} - Q promise object
  */
-function getTrunkData (configurationFile, extension) {
-  return dal.getSharedExtension(configurationFile, extension)
-    .then(function(sharedExtension) {
-      return sharedExtension.getTrunk(extension).then(function(trunk) {
-      return Q.resolve(trunk);
-    })
-    .catch(function(invalid) {
-      // This will only be called if trunk and extension are not the same
-      // Shouldn't be right now
-      return Q.reject(new customError.CustomError('InvalidTrunkSpecification',
-          invalid));
-    });
-  })
-  .catch(function(invalid) {
-    if (invalid.name === 'InvalidSpecification') {
-      return Q.reject(invalid);
+function isStation (extension, channel) {
+  var stations = extension.getAllStations();
+  var isItStation = false;
```

Comment:
isItStation is rather awkward. Since this is the result of the function itself you can simply call it result.